src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -589,27 +589,14 @@ public String getDuplicateNameCandidate(UUID directoryUuid, String elementName,
         return nameCandidate(elementName, i);
     }
 
-    public List<DirectoryElementInfos> searchElements(@NonNull String userInput) {
-        return directoryElementInfosService.searchElements(userInput)
-                    .stream()
-                    .map(this::populatePathInfo)
-                    .filter(Objects::nonNull)
-                    .toList();
+    public List<DirectoryElementInfos> searchElements(@NonNull String userInput, String directoryUuid) {
+        return directoryElementInfosService.searchElements(userInput, directoryUuid)
+                .stream().filter(element -> isElementExists(element.getParentId())) // filter the Orphan elements
```

Comment:
@SlimaneAmar in case of deletion of a directory , we can still have Orphans
```
//                              root
        //                    /                           \
        // dir1 (deleted but keeping its sub-elements)     dir2
        directoryElementRepository.deleteById(subDirUuid1);`

```